src/main/java/net/sf/jabref/logic/importer/MrDLibFetcher.java

```diff
@@ -0,0 +1,318 @@
+/**
+ *
+ */
+package net.sf.jabref.logic.importer;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.net.ssl.SSLContext;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import net.sf.jabref.model.entry.BibEntry;
+
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContexts;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+/**
+ *
+ *
+ */
+public class MrDLibFetcher implements SearchBasedFetcher {
+
+    private static final String NAME = "Mr. DLib Fetcher";
+    private List<BibEntry> bibEntryList;
+    private final BibEntry selectedEntry;
+    private List<String> htmlSnippets;
+    private static final Log LOGGER = LogFactory.getLog(MrDLibFetcher.class);
+
+
+    public MrDLibFetcher(BibEntry selectedEntry) throws Exception {
+        this.selectedEntry = selectedEntry;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public List<BibEntry> performSearch(String query) throws FetcherException {
```

Comment:
Strictly speaking, this is not really implementing the interface, as the query is being ignored. Instead of putting the entry into the constructor (and then reading its title from it), you can leave the constructor empty and push the latex-free title of the entry into this method. This would mean a slight and simple change on the side of `MrDlibFetcherWorker`.